app/jobs/appointment_reminders_job.rb

```diff
@@ -0,0 +1,28 @@
+class AppointmentRemindersJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    Appointment.within_48_hours.each do |appointment|
+      send_reminder(appointment)
+    end
+  end
+
+  private
+
+  def send_reminder(appointment)
+    expected_reminder_count = appointment.audits.inject(0) do |count, audit|
+      keys = audit.audited_changes.keys
+      keys.include?('start_at') ? count + 1 : count
+    end
+    actual_reminder_count = ReminderActivity.where(appointment: appointment).count
+
+    return if actual_reminder_count == expected_reminder_count
+
+    AppointmentMailer.reminder(appointment).deliver_later
+    ReminderActivity.create!(
```

Comment:
We really ought to do these the other way around since the mailer is called asynchronously (and has chance to recover from failures) but the persistence code doesn't.